### ISSUE TITLE
MapboxModuleProvider: Module loading failed in Android 12

### ISSUE THREAD

**Issue by HraD** [mapbox/mapbox-navigation-android#4967]
**Android API: 31**
**Device: Pixel 3a XL**
**Mapbox Navigation SDK version: com.mapbox.navigation:android:2.0.0-rc.4** (Any 2.x.x version)

### Steps to trigger behavior
Compile and run just any of navigation examples with "compileSdkVersion 31" and "targetSdkVersion 31".
Start example and got fatal exeption during mapbox navigation init:

```
            val navbuilder = NavigationOptions.Builder(requireContext())
                .accessToken(getMapboxAccessTokenFromResources())
                .deviceProfile(DeviceProfile.Builder().deviceType(DeviceType.AUTOMOBILE).build())
                .isFromNavigationUi(true)
                .isDebugLoggingEnabled(true)
                .navigatorPredictionMillis(2L)
             val mapboxNavigation = MapboxNavigation(navbuilder.build())
```
If you set "targetSdkVersion 30" - all works fine

### Expected behavior
All works fine

### Actual behavior
E/MapboxModuleProvider: Module loading failed
    java.lang.NoSuchMethodException: com.mapbox.navigation.trip.notification.internal.MapboxTripNotification.<init> []
        at java.lang.Class.getConstructor0(Class.java:2363)
        at java.lang.Class.getConstructor(Class.java:1759)
        at com.mapbox.common.module.provider.MapboxModuleProvider$noArgConstructorCreator$1.getInstance(MapboxModuleProvider.kt:107)
        at com.mapbox.common.module.provider.MapboxModuleProvider.createModule(MapboxModuleProvider.kt:69)
        at com.mapbox.navigation.core.MapboxNavigation.<init>(MapboxNavigation.kt:349)
        at ru.altatec.doraladigital.presentation.fragments.MultiLegRouteFragment.onViewCreated(MultiLegRouteFragment.kt:590)
        at androidx.fragment.app.Fragment.performViewCreated(Fragment.java:2987)
        at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:546)
        at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:282)
        at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2189)
        at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2100)
        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:2002)
        at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:524)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7842)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
E/MapboxModuleProvider: Module loading failed
    java.lang.NoSuchFieldException: INSTANCE
        at java.lang.Class.getField(Class.java:1635)
        at com.mapbox.common.module.provider.MapboxModuleProvider$kotlinObjectReferenceProvider$1.getInstance(MapboxModuleProvider.kt:117)
        at com.mapbox.common.module.provider.MapboxModuleProvider.createModule(MapboxModuleProvider.kt:69)
        at com.mapbox.navigation.core.MapboxNavigation.<init>(MapboxNavigation.kt:349)
        at ru.altatec.doraladigital.presentation.fragments.MultiLegRouteFragment.onViewCreated(MultiLegRouteFragment.kt:590)
        at androidx.fragment.app.Fragment.performViewCreated(Fragment.java:2987)
        at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:546)
        at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:282)
        at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2189)
        at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2100)
        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:2002)
        at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:524)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7842)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
E/MapboxModuleProvider: Module loading failed
    java.lang.NoSuchMethodException: com.mapbox.navigation.trip.notification.internal.MapboxTripNotification.getInstance []
        at java.lang.Class.getMethod(Class.java:2103)
        at java.lang.Class.getMethod(Class.java:1724)
        at com.mapbox.common.module.provider.MapboxModuleProvider$singletonReferenceProvider$1.getInstance(MapboxModuleProvider.kt:126)
        at com.mapbox.common.module.provider.MapboxModuleProvider.createModule(MapboxModuleProvider.kt:69)
        at com.mapbox.navigation.core.MapboxNavigation.<init>(MapboxNavigation.kt:349)
        at ru.altatec.doraladigital.presentation.fragments.MultiLegRouteFragment.onViewCreated(MultiLegRouteFragment.kt:590)
        at androidx.fragment.app.Fragment.performViewCreated(Fragment.java:2987)
        at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:546)
        at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:282)
        at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2189)
        at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2100)
        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:2002)
        at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:524)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7842)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
E/MapboxModuleProvider: Module loading failed
    java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Constructor.newInstance0(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
        at com.mapbox.common.module.provider.MapboxModuleProvider$defaultMapboxModuleCreator$1.getInstance(MapboxModuleProvider.kt:138)
        at com.mapbox.common.module.provider.MapboxModuleProvider.createModule(MapboxModuleProvider.kt:69)
        at com.mapbox.navigation.core.MapboxNavigation.<init>(MapboxNavigation.kt:349)
        at ru.altatec.doraladigital.presentation.fragments.MultiLegRouteFragment.onViewCreated(MultiLegRouteFragment.kt:590)
        at androidx.fragment.app.Fragment.performViewCreated(Fragment.java:2987)
        at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:546)
        at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:282)
        at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2189)
        at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2100)
        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:2002)
        at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:524)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7842)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
     Caused by: java.lang.IllegalArgumentException: ru.altatec.doraladigital: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
    Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
        at android.app.PendingIntent.checkFlags(PendingIntent.java:375)
        at android.app.PendingIntent.getActivityAsUser(PendingIntent.java:458)
        at android.app.PendingIntent.getActivity(PendingIntent.java:444)
        at android.app.PendingIntent.getActivity(PendingIntent.java:408)
        at com.mapbox.navigation.trip.notification.internal.MapboxTripNotification.createPendingOpenIntent(MapboxTripNotification.kt:220)
        at com.mapbox.navigation.trip.notification.internal.MapboxTripNotification.<init>(MapboxTripNotification.kt:98)
        	... 21 more
D/AndroidRuntime: Shutting down VM
    
    
    --------- beginning of crash
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: ru.altatec.doraladigital, PID: 9714
    MapboxInvalidModuleException(type=NavigationTripNotification)
        at com.mapbox.common.module.provider.MapboxModuleProvider.createModule(MapboxModuleProvider.kt:88)
        at com.mapbox.navigation.core.MapboxNavigation.<init>(MapboxNavigation.kt:349)
        at ru.altatec.doraladigital.presentation.fragments.MultiLegRouteFragment.onViewCreated(MultiLegRouteFragment.kt:590)
        at androidx.fragment.app.Fragment.performViewCreated(Fragment.java:2987)
        at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:546)
        at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:282)
        at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2189)
        at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2100)
        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:2002)
        at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:524)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7842)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)

D/nav-native: PersistentConfig loading started
D/nav-native: PersistentConfig loaded
I/nav-native: Async config init done with version 2021_10_02-03_00_00 [{"map":{"tileset_version":"2021_10_02-03_00_00"}}]

**Comment by dudeuter** [comment]
Hello @HraD,

I believe you should be able to resolve this by upgrading to a more recent release of the Navigation SDK 2.0.

@sjd753 May I inquire on which version of the SDK you are building with?

Also, if either of you have not already it might be best to create a Support Request at https://support.mapbox.com/hc/en-us. If you have already then you might either want to share your ticket number with me, or create a new request with the same email and account. 🙂

**Comment by LukasPaczos** [comment]
Highlighting the relevant portion of the stack:
```
Caused by: java.lang.IllegalArgumentException: ru.altatec.doraladigital: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
at android.app.PendingIntent.checkFlags(PendingIntent.java:375)
at android.app.PendingIntent.getActivityAsUser(PendingIntent.java:458)
at android.app.PendingIntent.getActivity(PendingIntent.java:444)
at android.app.PendingIntent.getActivity(PendingIntent.java:408)
at com.mapbox.navigation.trip.notification.internal.MapboxTripNotification.createPendingOpenIntent(MapboxTripNotification.kt:220)
at com.mapbox.navigation.trip.notification.internal.MapboxTripNotification.(MapboxTripNotification.kt:98)
```

**Comment by LukasPaczos** [comment]
Closing as the fix is available in `v2.0.2`, `v2.1.0-rc.2`, and will be available in the upcoming `v1.6.2`. Thanks for reporting the issue!
